### PR TITLE
Ensure posts are sorted chronologically

### LINKED ISSUE #99: RSS feed is out of order
Items in the [published RSS feed](https://nodejs.github.io/nodejs-news-feeder/feed.xml) are non-chronological, which is causing an issue for me with an RSS<>Discord connector that seems to presume that the first post will always be the most recent

Screenshot 2024-07-29 at 2 30 25 PM

### PR DESCRIPTION
**Pull request by vcarl**
Resolves #99

### CODE REVIEW

File: scripts/format.js [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+
+const parser = new ParseRss()
 
 const xml = getFeedContent()
-const formattedXml = xmlFormat(xml, { indentation: '  ', collapseContent: true })
-overwriteFeedContent(formattedXml)
+
+parser.parseString(xml).then((parsedXml) => {
+  const sortedItems = parsedXml.items
+    .sort((a, b) => (new Date(a.isoDate) < new Date(b.isoDate) ? 1 : -1))
```
Comment (ovflowd) [inline scripts/format.js:18]: You can do new Date(a.isoDate) - new Date(b.isoDate)

File: scripts/format.js [hunk 2]
```diff
@@ -1,6 +1,32 @@
...
+    .map(composeFeedItem)
+    .join('')
+
+  const feedContent = getFeedContent()
```
Comment (ovflowd) [inline scripts/format.js:22]: Why we call getFeedContent twice?

**Review by ovflowd — APPROVED** [review]
LGTM! Left a few comments!

File: scripts/format.js (hunk 2 above)
Comment (vcarl) [reply scripts/format.js:22]: Whoop def unintentional, tyty

File: scripts/format.js (hunk 1 above)
Comment (vcarl) [reply scripts/format.js:18]:
```suggestion
    .sort((a, b) => new Date(a.isoDate) - new Date(b.isoDate))
```

File: scripts/format.js [hunk 3]
```diff
@@ -1,6 +1,32 @@
...
+  const [before] = feedContent.split(breakDelimiter)
```
Comment (vcarl) [inline scripts/format.js:23]:
```suggestion
  const [before] = xml.split(breakDelimiter)
```

**Comment by vcarl** [comment]
Yep just addressing those!

**Comment by UlisesGascon** [comment]
Not sure why the rss checker is complaining :thinking: 
Screenshot from 2024-08-22 21-16-58

You can simulate this by running: `npm run build && npm run rss:format-check`, and then if you run `npm run rss:format` you can see what have changed

**Review by UlisesGascon — APPROVED** [review]
Thanks @vcarl for all the work! :heart:

**Comment by UlisesGascon** [comment]
So... this change was deployed and also I included a content update recently. Let me know if you find any issue @vcarl :+1:

**Pull request merged**